src/frontend/js/modules/session.js

```diff
@@ -92,30 +94,45 @@
     $scope.domain = session.domain;
   })
 
+  // TODO (esn-frontend-common-libs#51): Write tests for the new changes (https://github.com/OpenPaaS-Suite/esn-frontend-common-libs/pull/48)
   .controller('sessionInitESNController', function($scope, esnTemplate, sessionFactory) {
-
     $scope.session = {
       template: esnTemplate.templates.loading
     };
 
-    sessionFactory.fetchUser(function(error) {
-      if (error) {
+    sessionFactory.bootstrapSession()
+      .then(() => {
+        $scope.session.template = esnTemplate.templates.success;
+      })
+      .catch(error => {
         $scope.session.error = error.data;
         $scope.session.template = esnTemplate.templates.error;
-      } else {
-        $scope.session.template = esnTemplate.templates.success;
-      }
-    });
+      });
   })
 
-  .factory('sessionFactory', function($log, $q, Restangular, userAPI, domainAPI, session) {
+  .factory('sessionFactory', function($log, $q, Restangular, userAPI, domainAPI, session, themesService, applyThemeService) {
 
     function onError(error, callback) {
           if (error && error.data) {
             return callback(error.data);
           }
         }
 
+    function bootstrapSession() {
+      return new Promise((resolve, reject) => {
+        fetchUser(error => {
+          if (error) return reject(error);
+
+          themesService.getTheme(session.domain._id).then(theme => {
+            applyThemeService.applyTheme(theme);
+            resolve();
+          }).catch(error => {
+            reject(error);
+          });
+        });
+      });
+    }
```

Comment:
We will have to think if it is better to to 2 HTTP requests to initialize the so-called `session` (and potentially more than 2), or to have an endpoint which sends back everything needed, or having the `user` object which contains everything needed

Reply:
Good point. I think there can be only one endpoint that sends back everything needed for bootstraping a session. Wdyt @tuanlc @MichaelBailly?

Reply:
Yes, this is an improvement that you'll do